src/Chapter1.hs

```diff
@@ -554,7 +558,11 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | (x > y && x < z) || (x < y && x > z) = x
+    | (y > x && y < z) || (y < x && y > z) = y
+    | (z > x && z < y) || (z < x && z > y) = z
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = z`, to tell the compiler, that your pattern matching is exhaustive 🙂


It looks like you can replace the last case with a single otherwise and return z in here.
The implementation will be a bit shorter 🙂

Also, I think that you can use `>=` and `<=` operators instead of `<` and `>` to cover cases when some of the arguments are equal.